python/cuml/test/test_pickle.py

```diff
@@ -18,50 +18,44 @@
 import pickle
 import pytest
 
-from cuml.test.utils import array_equal, unit_param, stress_param
+from cuml.test.utils import array_equal, unit_param, stress_param, \
+    ModuleConfig
 from cuml.test.test_svm import compare_svm
 from sklearn.base import clone
-from sklearn.datasets import load_iris, make_classification,\
-    make_regression
+from sklearn.datasets import load_iris, make_classification, make_regression
 from sklearn.manifold.t_sne import trustworthiness
 from sklearn.model_selection import train_test_split
 
-regression_models = {
-    "LinearRegression": lambda fit_intercept=True: cuml.LinearRegression(
-        fit_intercept=fit_intercept),
-    "LogisticRegression": lambda fit_intercept=True: cuml.LogisticRegression(
-        fit_intercept=fit_intercept),
-    "Lasso": lambda fit_intercept=True: cuml.Lasso(
-        fit_intercept=fit_intercept),
-    "Ridge": lambda fit_intercept=True: cuml.Ridge(
-        fit_intercept=fit_intercept),
-    "ElasticNet": lambda fit_intercept=True: cuml.ElasticNet(
-        fit_intercept=fit_intercept)
-}
-
-solver_models = {
-    "CD": lambda: cuml.CD(),
-    "SGD": lambda: cuml.SGD(eta0=0.005),
-    "QN": lambda: cuml.QN(loss="softmax")
-}
-
-cluster_models = {
-    "KMeans": lambda: cuml.KMeans()
-}
-
-decomposition_models = {
-    "PCA": lambda: cuml.PCA(),
-    "TruncatedSVD": lambda: cuml.TruncatedSVD(),
-}
-
-decomposition_models_xfail = {
-    "GaussianRandomProjection": lambda: cuml.GaussianRandomProjection(),
-    "SparseRandomProjection": lambda: cuml.SparseRandomProjection()
-}
-
-neighbor_models = {
-    "NearestNeighbors": lambda: cuml.NearestNeighbors()
-}
+regression_config = ModuleConfig(module=cuml.linear_model)
+regression_models = regression_config.get_models()
+
+solver_config = ModuleConfig(
+    module=cuml.solvers,
+    custom_constructors={
+        "SGD": lambda: cuml.SGD(eta0=0.005),
+        "QN": lambda: cuml.QN(loss="softmax")
+    }
+)
+solver_models = solver_config.get_models()
+
+cluster_config = ModuleConfig(
+    module=cuml.cluster,
+    exclude_classes=[cuml.DBSCAN]
+)
+cluster_models = cluster_config.get_models()
+
+decomposition_config = ModuleConfig(module=cuml.decomposition)
+decomposition_models = decomposition_config.get_models()
+
+decomposition_config_xfail = ModuleConfig(module=cuml.random_projection)
+decomposition_models_xfail = decomposition_config_xfail.get_models()
+
+neighbor_config = ModuleConfig(
+    module=cuml.neighbors,
+    exclude_classes=[cuml.neighbors.KNeighborsClassifier,
+                     cuml.neighbors.KNeighborsRegressor]
```

Comment:
PR for solving this is in, so no need to exclude

Reply:
Correct me if I am wrong but I think we do need to exclud them here because they are in fact tested later in:
https://github.com/rapidsai/cuml/blob/6c102c0b15a3fc29eb8654fb1ac04a57f7d036a5/python/cuml/test/test_pickle.py#L71-L73
The tests for **neighbors** and **k_neighbors** are slightly different.

Reply:
Same comment as @JohnZed mentioned applies here, `k_neighbors` should also pass the general with xfail if needed (which I presume shouldn't be needed)

Reply:
Fair enough. Working on it.